source/neuropod/bindings/java/src/main/native/com_uber_neuropod_NeuropodTensorAllocator.cc

```diff
@@ -0,0 +1,35 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "com_uber_neuropod_NeuropodTensorAllocator.h"
+
+#include "neuropod/neuropod.hh"
+#include "utils.h"
+
+#include <jni.h>
+
+JNIEXPORT void JNICALL Java_com_uber_neuropod_NeuropodTensorAllocator_nativeDelete(JNIEnv *env, jobject, jlong handle)
+{
+    try
+    {
+        auto allocator = reinterpret_cast<std::shared_ptr<neuropod::NeuropodTensorAllocator> *>(handle);
+        (*allocator).reset();
+        delete allocator;
```

Comment:
delete 0 or delete nullptr is allowed and safe. But not *nullptr. I guess it should make sure we don't dereference nullptr here:

```
if (allocator != nullptr) {
        allocator->reset();
        delete allocator;
}
```

Reply:
In the java side close method, we have if (nativeHandle == 0) { return;} to ensure the handle is non zero.

Reply:
nativeDelete method is only called by close method, so this check always applies.